File: MuayeneYonetimPortali/MuayeneYonetimPortali.Web/Modules/ServerTypes/Tanimlamalar/CityRow.ts
import { fieldsProxy } from "@serenity-is/corelib";

export interface CityRow {
    CityId?: number;
    Name?: string;
}

export abstract class CityRow {
    static readonly idProperty = 'CityId';
    static readonly nameProperty = 'Name';
    static readonly localTextPrefix = 'Tanimlamalar.City';
    static readonly deletePermission = 'Tanimlamalar.City.General';
    static readonly insertPermission = 'Tanimlamalar.City.General';
    static readonly readPermission = 'Tanimlamalar.City.General';
    static readonly updatePermission = 'Tanimlamalar.City.General';

    static readonly Fields = fieldsProxy<CityRow>();
}

File: MuayeneYonetimPortali/MuayeneYonetimPortali.Web/Modules/Tanimlamalar/City/CityRow.cs
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System.ComponentModel;

namespace MuayeneYonetimPortali.Tanimlamalar;

[ConnectionKey("Default"), Module("Tanimlamalar"), TableName("City")]
[DisplayName("City"), InstanceName("City")]
[ReadPermission("Tanimlamalar.City.General")]
[ModifyPermission("Tanimlamalar.City.General")]
[ServiceLookupPermission("Tanimlamalar.City.General")]
public sealed class CityRow : Row<CityRow.RowFields>, IIdRow, INameRow
{
    [DisplayName("City Id"), Identity, IdProperty]
    public int? CityId { get => fields.CityId[this]; set => fields.CityId[this] = value; }

    [DisplayName("Name"), Size(200), NotNull, QuickSearch, NameProperty]
    public string Name { get => fields.Name[this]; set => fields.Name[this] = value; }

    public class RowFields : RowFieldsBase
    {
        public Int32Field CityId;
        public StringField Name;

    }
}